### DISCUSSION TITLE
What's the least NodeJS version should Rspack support ?

### DISCUSSION THREAD

**Discussion by hardfist** [web-infra-dev/rspack#4579] [Ideas]
Why is NodeJS 14 required for the pipeline?

_Originally posted by @hverlin in https://github.com/web-infra-dev/rspack/issues/4509#issuecomment-1802892467_

we need to decide the least NodeJS version we should support before release 1.0, otherwise we will be stuck it for years and we can't use the latest tools in our repo(like pnpm 8) cause every NodeJS upgrade is a breaking change.
currently NodeJS 16 is also EOL and vite@5 will drop NodeJS 16 support and only support Node18 and Node 20+ https://main.vitejs.dev/guide/migration#migration-from-v4

**Comment by hverlin** [comment]
Are there any reasons to support NodeJS 14 (given that it is EOL, https://nodejs.org/en/about/previous-releases)?
`rspack` is likely only used on the frontend, where it's usually much easier to upgrade NodeJS version (than on the backend)

**Reply by hardfist** [reply]
some of our internal users run sever and build on same environment，so it is not as easy to upgrade server environment

**Comment by hardfist** [comment]
we will deprecate Node14 support since 0.4.0 and continue support Node16 for some time
